### PR TITLE
feat: Remove blog duplicate code

### PR DESCRIPTION
**Pull request by ChefMomota**
### Summary
🎮📦🎨

This pull request adds a new feature for games articles and refactors the blog app code. It introduces a new dependency on `@pancakeswap/blog` to use the shared types and utils for the blog data. It also reorganizes and sorts the imports in various files and consolidates the article components in one file. It deletes some unused files and updates the pnpm version and lock file.

> _The blog app has a new feature_
> _For games articles with some allure_
> _The imports are sorted_
> _The code is refactored_
> _And `@pancakeswap/blog` is the cure_

### Walkthrough
*  Refactor article components to a single file `./apps/blog/components/Article/AllArticle/index.tsx` and reorder imports alphabetically and by source ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-90fe91a04d505f100c27729fc4fb98d49e286574aabc0961a3db098868d534bcL1-R14), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-66857c962e223b2cf291bdbce5015ebd31c69a49f493522e12d1b1c90d9645e4L1-R5), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-bb7a12c1cd1186cb70a9d06c42bd891c0f2cfae41a911b7594ab5ec35350cb5bL1-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-1642e734ef393fc2a9f34306c08689d507c2329e7f5eb5ef665f02905f2e900aL1-R7), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-083a28cd9147f41f63134a2d8a062f5fc2699d5bf413b5d0c58682fb2f81698dL1-R12), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-f392ae424e2c767c7923543333ddedfda815753f242fb7fc2b7cb46655177046L1-R4), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-2d069d12b02fc9c8b5723fb2ca46ddc19ff506ecad9f1e4575b206c2fd56011cL1-R12), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-baf402b322f2e0097407683b674cb473ee2c2429b7a9a7cf334128a94ee4caedL1-R8))
* Add `@pancakeswap/blog` package as a dependency and use its shared types and utils in the blog app ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-9a5b1aec248cd6dede355176954b5c3da29f45cd6f8e335d508618c12de5a933L1-R10), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-c85a123e12f6ce3312acc4fa465eb1f711a774b02c1b17fc6f85cd5c96521c32L1-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-4d6a255b06cab3f3aad9f6e18cdf49d404b8dd6f177f71f059ae1a23a9c181f8L1-R3), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-cd29b0763a1190c00f640a32386d0e2b0c03f9ba4701d55e71abc1cb607d0171R16), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbR370-R372))
* Add `gamesCategories` property to the article data to support a new category for games articles ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-9a5b1aec248cd6dede355176954b5c3da29f45cd6f8e335d508618c12de5a933R58), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-0961bf0632731313ad57818d6f9473f6824f13fe0a4074881dd5fac4c654b576L38-R38))
* Reorder imports alphabetically and by source in other files ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-956a5360205f9116704a061c4caf856a85557ef36b0d9e99469fc249f09c0da6L1-R12), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-0961bf0632731313ad57818d6f9473f6824f13fe0a4074881dd5fac4c654b576L1-R1))
* Update `lockfileVersion` in `./pnpm-lock.yaml` to match the pnpm version used in the project ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-32824c984905bb02bc7ffcef96a77addd1f1602cff71a11fbbfdd7f53ee026bbL1-R1))
* Delete unused files `apps/blog/types.ts` and `apps/blog/utils/transformArticle.ts` ([link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-be86497777a3b318bd86840a83b8e55239456255e2cb4a24e33da363cb90304c), [link](https://github.com/pancakeswap/pancake-frontend/pull/8562/files?diff=unified&w=0#diff-dd5151d886e1c520d3df4db0026868c1ebe300ced3a0997af9c61043b7e9f448))

### CODE REVIEW

File: pnpm-lock.yaml [hunk 1]
```diff
@@ -1,4 +1,4 @@
-lockfileVersion: '6.0'
+lockfileVersion: '6.1'
```
Comment (chefjackson) [inline pnpm-lock.yaml:1]: hmm weird may need to take a look what's the difference between 6.1 and 6.0

Comment (ChefMomota) [reply pnpm-lock.yaml:1]: In my local when I run `pnpm i` it will auto change to `6.1`

Comment (chefjackson) [reply pnpm-lock.yaml:1]: which pnpm version are you using?

Comment (ChefMomota) [reply pnpm-lock.yaml:1]: `8.6.0`

Comment (chefjackson) [reply pnpm-lock.yaml:1]: please upgrade to the latest

**Review by chefjackson — APPROVED** [review]

**Pull request merged**